packages/authentication/node-tests/authentication-test.js

```diff
@@ -31,6 +31,16 @@ describe('authentication/middleware', function() {
         mayReadFields: true
       });
 
+    factory.addResource('grants')
+      .withRelated('who', { type: 'groups', id: 'everyone' })
+      .withRelated('types', [
+        { type: 'content-types', id: 'test-users' },
+        { type: 'content-types', id: 'doggies' }
+      ])
```

Comment:
This is a grant to provide `mayLogin` capabilities to all users whose session is described by a `test-users` content type or a `doggies` content type. Adding a `mayLogin` authorization to the grant above doesn't work for some of the token issuers tests as they belong to realms that are no covered in the grant above because of the  `.withRelated('who', { type: 'fields', id: 'id' })`

Reply:
I could leave out the types, and that would work too. but eventually when we implement #181, we'll need to probably do something like this.

Reply:
I don't think #181 applies unless we want to be able to limit mayLogin grants based on user type, and I don't think we actually need that feature. We can defer that kind of capability until we have a more complete group membership system, because I think that will cover it equally well.

Reply:
In any case, overloading the `types` field seems risky to me. If a grant has both `mayLogin` and `mayWriteResource`, the `types` field is doing double-duty in a way that may troll people.

Reply:
sounds good